### PR TITLE
Add kedro-mlflow to the list of community plugins (#113)

### PR DESCRIPTION
**Pull request by Galileo-Galilei**
## Description

This PR aims at keeping a [long-standing promise to @lorenabalan and @yetudada](https://github.com/quantumblacklabs/kedro/issues/113#issuecomment-704214620) by adding [kedro-mlflow](https://github.com/Galileo-Galilei/kedro-mlflow) to the list of community developped plugins.

## Development notes
Add a (not so?) brief description of what the plugin does in documentation.

## Checklist

- [X] Read the [contributing](https://github.com/quantumblacklabs/kedro/blob/master/CONTRIBUTING.md) guidelines
- N/A Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- N/A Updated the documentation to reflect the code changes
- N/A Added a description of this change and added my name to the list of supporting contributions in the [`RELEASE.md`](https://github.com/quantumblacklabs/kedro/blob/master/RELEASE.md) file
- N/A Added tests to cover my changes

## Notice

- [X] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

- I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
- I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorised to submit this contribution on behalf of the original creator(s) or their licensees.
- I certify that the use of this contribution as authorised by the Apache 2.0 license does not violate the intellectual property rights of anyone else.

### CODE REVIEW

**Review by limdauto — APPROVED** [review]
LGTM!

**Review by yetudada — APPROVED** [review]
You don't know how happy this makes me 🎉 Thank you so much @Galileo-Galilei!

File: docs/source/07_extend_kedro/04_plugins.md [hunk 1]
```diff
@@ -149,3 +149,4 @@ When you are ready to submit your code:
...
 - [kedro-kubeflow](https://github.com/getindata/kedro-kubeflow), by [Mateusz Pytel](https://github.com/em-pe) and [Mariusz Strzelecki](https://github.com/szczeles), lets you run and schedule pipelines on Kubernetes clusters using [Kubeflow Pipelines](https://www.kubeflow.org/docs/pipelines/overview/pipelines-overview/)
+- [kedro-mlflow](https://github.com/Galileo-Galilei/kedro-mlflow), by [Yolan Honoré-Rougé](https://github.com/galileo-galilei), [Kajetan Maurycy Olszewski](https://github.com/kaemo), and [Takieddine Kadiri](https://github.com/takikadiri) facilitates [Mlflow](https://www.mlflow.org/) integration inside Kedro projects while enforcing [Kedro's principles](https://kedro.readthedocs.io/en/stable/12_faq/01_faq.html?highlight=principles#what-is-the-philosophy-behind-kedro). Its main features are modular configuration, automatic parameters tracking, datasets versioning, Kedro pipelines packaging and serving and automatic synchronization between training and inference pipelines for high reproducibility of machine learning experiments and ease of deployment. A tutorial is provided in the [kedro-mlflow-tutorial repo](https://github.com/Galileo-Galilei/kedro-mlflow-tutorial).
```
Comment (lorenabalan) [inline docs/source/07_extend_kedro/04_plugins.md:152]: Did you maybe mean this one? The previous link didn't exist.

```suggestion
- [kedro-mlflow](https://github.com/Galileo-Galilei/kedro-mlflow), by [Yolan Honoré-Rougé](https://github.com/galileo-galilei), [Kajetan Maurycy Olszewski](https://github.com/kaemo), and [Takieddine Kadiri](https://github.com/takikadiri) facilitates [Mlflow](https://www.mlflow.org/) integration inside Kedro projects while enforcing [Kedro's principles](https://kedro.readthedocs.io/en/stable/12_faq/01_faq.html?highlight=principles#what-are-the-primary-advantages-of-kedro). Its main features are modular configuration, automatic parameters tracking, datasets versioning, Kedro pipelines packaging and serving and automatic synchronization between training and inference pipelines for high reproducibility of machine learning experiments and ease of deployment. A tutorial is provided in the [kedro-mlflow-tutorial repo](https://github.com/Galileo-Galilei/kedro-mlflow-tutorial).
```

Should make linkchecker pass.

**Review by lorenabalan — APPROVED** [review]
Fantastic!! 👏

File: docs/source/07_extend_kedro/04_plugins.md (hunk 1 above)
Comment (Galileo-Galilei) [reply docs/source/07_extend_kedro/04_plugins.md]: Well spotted! I have never used the CI to check for broken links but the idea is so cool that I added it to kedro mlflow (and discover dozens of broken links in the docs...). Definitely worth opening this PR, I've learnt something today :)

I've just rebased/squashed and pushed again, it should be ok now.

**Pull request merged**